src/encoding.ts

```diff
@@ -188,7 +188,7 @@ export interface Encoding<F extends Field> {
   color?:
     | ColorGradientFieldDefWithCondition<F>
     | ColorGradientDatumDefWithCondition<F>
-    | ColorGradientValueOrSignalDefWithCondition<F>;
```

Comment:
I didn't like the fact that we have to do `ColorGradientValueOrSignalDefWithCondition` because once we expose signal as an official feature, value and signal will show up together in the docs.  

If want to keep signal def but separate value and signal in the property table, then we have to declare so many repetitive types.
For every variant of value or signal def (e.g., `ColorGradient`), we have to do:

```
XXXValueDefWithCondition
XXXSignalDefWithCondition
```